src/main/java/eu/stamp_project/mutationtest/descartes/operators/ThisMutationOperator.java

```diff
@@ -22,8 +22,8 @@
     @Override
     public boolean canMutate(ClassName className, Method method) {
         String classNameAsInternal = className.asInternalName();
-        String methodReturn = method.getReturnType().toString();
-        if (methodReturn.contains(classNameAsInternal)) {
+        String methodReturn = method.getReturnType().getInternalName();
+        if (classNameAsInternal.equals(methodReturn)) {
```

Comment:
`return classNameAsInternal.equals(methodReturn)`

Reply:
Updated the function! Sorry for the same.